src/history/hs_rec.c

```diff
@@ -159,33 +162,29 @@ __hs_insert_record_with_btree_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor, u
  *     Should be called with session's btree switched to the history store.
  */
 static int
-__hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
-  const WT_ITEM *key, const uint8_t type, const WT_ITEM *hs_value,
-  WT_HS_TIME_POINT *start_time_point, WT_HS_TIME_POINT *stop_time_point)
+__hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_HS_KEY *hs_key,
+  WT_HS_VALUE *hs_value, WT_TIME_WINDOW *tw)
 {
 #ifdef HAVE_DIAGNOSTIC
     WT_CURSOR_BTREE *hs_cbt;
 #endif
-    WT_DECL_ITEM(hs_key);
     WT_DECL_ITEM(srch_key);
 #ifdef HAVE_DIAGNOSTIC
     WT_DECL_ITEM(existing_val);
 #endif
     WT_DECL_RET;
+    WT_HS_KEY retrieved_key;
     wt_timestamp_t hs_start_ts;
 #ifdef HAVE_DIAGNOSTIC
     wt_timestamp_t durable_timestamp_diag;
     wt_timestamp_t hs_stop_durable_ts_diag;
     uint64_t upd_type_full_diag;
 #endif
-    uint64_t counter, hs_counter;
-    uint32_t hs_btree_id;
     int cmp;
 
-    counter = 0;
-
+    hs_key->counter = 0;
     /* Allocate buffers for the history store and search key. */
-    WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
+    WT_ERR(__wt_scr_alloc(session, 0, &retrieved_key.key));
```

Comment:
I'm not a huge fan of `retrieved_key.key`, would it be better to rename `key` to `ds_key` in the `WT_HS_KEY` struct?

Reply:
agree, that is better.